### PR TITLE
add docker image

### PR DESCRIPTION
**Pull request by Tianhao-Gu**

### CODE REVIEW

File: src/loaders/jobs/docker_img/scripts/compute_genome_entrypoint.sh [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
+# Check if conda is available
+elif command -v conda >/dev/null 2>&1; then
+  echo "conda found, using conda"
+  conda_cmd="conda"
+else
+  echo "Neither conda nor micromamba found. Please install either conda or micromamba."
+  exit 1
+fi
+
+$conda_cmd run -n $CONDA_ENV \
```
Comment (MrCreosote) [inline src/loaders/jobs/docker_img/scripts/compute_genome_entrypoint.sh:16]: I think it would be fine to have a custom entrypoint script for tools when needed rather than having one script with a bunch of `if else` blocks.

Or alternatively, make `$conda_cmd` an environment variable defaulting to `conda` and set that env var to `micromamba` in the microtrait docker file.

File: src/loaders/jobs/docker_img/Dockerfile.microtrait [hunk 2]
```diff
@@ -0,0 +1,36 @@
+FROM mambaorg/micromamba:1.4.1
+
+# Note from micromamba - using an environment name other than "base" is not recommended
+ENV CONDA_ENV base
+
+COPY --chown=$MAMBA_USER:$MAMBA_USER /src/loaders/jobs/docker_img/env_microtrait.yaml /tmp/env_microtrait.yaml
+RUN micromamba install -y -n $CONDA_ENV -f /tmp/env_microtrait.yaml && \
+    micromamba clean --all --yes
+
+USER root
```
Comment (MrCreosote) [inline src/loaders/jobs/docker_img/Dockerfile.microtrait:10]: Why is this necessary?

File: src/loaders/jobs/docker_img/Dockerfile.microtrait [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+# Install microTrait
+RUN Rscript -e 'install.packages("kmed", repos = "https://cloud.r-project.org/")' && \
+    Rscript -e 'devtools::install_github("jlw-ecoevo/gRodon")' && \
+    Rscript -e 'devtools::install_github("ukaraoz/microtrait")'
+
+# Setup of HMM database
+RUN Rscript -e 'library(microtrait)' && \
+    Rscript -e 'microtrait::prep.hmmmodels()' && \
+    Rscript -e 'list.files(file.path(.libPaths(), "microtrait/extdata/hmm/hmmpress"))'
```
Comment (MrCreosote) [inline src/loaders/jobs/docker_img/Dockerfile.microtrait:21]: Is this something that should be installed outside of the dockerfile like the gtdb and checkm databases?

File: src/loaders/jobs/docker_img/Dockerfile.microtrait [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+
+ARG MAMBA_DOCKERFILE_ACTIVATE=1
```
Comment (MrCreosote) [inline src/loaders/jobs/docker_img/Dockerfile.microtrait:23]: seems to be unused?

File: src/loaders/jobs/docker_img/Dockerfile.microtrait [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+RUN echo "source activate $CONDA_ENV" >> ~/.bashrc
```
Comment (MrCreosote) [inline src/loaders/jobs/docker_img/Dockerfile.microtrait:24]: Why does this need to go in `.bashrc` rather than `RUN`ing it?

File: src/loaders/jobs/docker_img/env_microtrait.yaml [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+  - r-tidyr
+  - bioconductor-biostrings
+  - bioconductor-cordon
+  - bioconductor-complexheatmap
+  - hmmer=3.3.2
+  - prodigal=2.6.3
+  - barrnap=0.9
+  - ipython
+  - rpy2
+  - pandas
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/env_microtrait.yaml]: 👍 being lazy.

File: src/loaders/jobs/docker_img/scripts/compute_genome_entrypoint.sh (hunk 1 above)
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/scripts/compute_genome_entrypoint.sh]: 👍

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 2 above)
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: 👍 removed.

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 3 above)
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:22]: the doc from microtrait didn't mention we could. It's not taking too much space either. I think it might not worth to read the R code and trying to figure out how hmm database is integrated with the microtrait tool. 
https://github.com/ukaraoz/microtrait/tree/master#5--setup-of-hmm-database

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 4 above)
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: 👍  removed.

File: src/loaders/jobs/docker_img/Dockerfile.microtrait [hunk 7]
```diff
@@ -0,0 +1,36 @@
+FROM mambaorg/micromamba:1.4.1
+
+# Note from micromamba - using an environment name other than "base" is not recommended
+ENV CONDA_ENV base
+
+COPY --chown=$MAMBA_USER:$MAMBA_USER /src/loaders/jobs/docker_img/env_microtrait.yaml /tmp/env_microtrait.yaml
```
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:8]: yes. it comes with the base image. And according to their doc, it's the recommended way to do COPY.
https://github.com/mamba-org/micromamba-docker#quick-start

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 5 above)
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: 👍 removed. not necessary. especially we are using `base`.

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 7 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:8]: Can you add a comment explaining that?

File: src/loaders/jobs/docker_img/env_microtrait.yaml (hunk 6 above)
[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/env_microtrait.yaml]: :+1:

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 2 above)
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: :+1:

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 3 above)
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:22]: If it's small then NBD

Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:22]: :+1:

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 4 above)
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: :+1:

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 5 above)
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait]: :+1:

File: src/loaders/jobs/docker_img/Dockerfile.microtrait (hunk 7 above)
[the remark this answers is not in the document]
Comment (Tianhao-Gu) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:11]: 👍

[the remark this answers is not in the document]
Comment (MrCreosote) [reply src/loaders/jobs/docker_img/Dockerfile.microtrait:11]: :+1:

**Review by MrCreosote — APPROVED** [review]
LGTM
